crates/cli/src/config.rs

```diff
@@ -33,10 +59,29 @@ impl ServerConfig {
     }
 }
 
-#[derive(Default, Deserialize, Serialize, Debug, Clone)]
+impl From<&Table> for ServerConfig {
+    fn from(table: &Table) -> Self {
+        let nickname = table.get(NICKNAME_KEY).and_then(Item::as_str).map(String::from);
+        let host = table.get(HOST_KEY).and_then(Item::as_str).map(String::from).unwrap();
+        let protocol = table
+            .get(PROTOCOL_KEY)
+            .and_then(Item::as_str)
+            .map(String::from)
+            .unwrap();
+        let ecdsa_public_key = table.get(ECDSA_PUBLIC_KEY).and_then(Item::as_str).map(String::from);
+        ServerConfig {
+            nickname,
+            host,
+            protocol,
+            ecdsa_public_key,
+        }
+    }
+}
+
+// Any change here must be coordinated with Config::doc
```

Comment:
Please expand this, and make it into a doc comment. Do you mean:

- Changes at the source level by SpacetimeDB developers to this type definition?
- Changes at runtime to the data contained in a `RawConfig` instance?

Who is responsible for the coordination? In our actual uses of this type, how do those changes propagate?